python_modules/dagster/dagster/_cli/api.py

```diff
@@ -135,10 +137,12 @@ def _execute_run_command_body(
         run_worker_failed = True
     finally:
         if instance.should_start_background_run_thread:
-            cancellation_thread_shutdown_event.set()
-            if cancellation_thread.is_alive():
-                cancellation_thread.join(timeout=15)
-                if cancellation_thread.is_alive():
+            shutdown_event = check.not_none(cancellation_thread_shutdown_event)
+            thread = check.not_none(cancellation_thread)
+            shutdown_event.set()
+            if thread.is_alive():
+                thread.join(timeout=15)
+                if thread.is_alive():
```

Comment:
This isn't the greatest naming pattern. `cancellation_thread_shutdown_event` --> `shutdown_event` sounds like we are entering a more generalized codepath, which is not true. This just needed a new name for typechecking purposes. 

For example, renaming the existing variable to `maybe_cancellation_thread_shutdown_event` and calling the post check one `cancellation_thread_shutdown_event` is much more accurate IMO, even though it is a little verbose. Open to other naming conventions, but not a big fan of the one in this PR.

Reply:
Renamed-- typing here does not actually require a new variable name since the type of the reassigned var is a subset of the old type.